src/test/java/io/fabric8/maven/docker/util/ImageNameTest.java

```diff
@@ -125,6 +125,8 @@ public void testRegistryNamingExtended() throws Exception {
                 new ImageName("docker.jolokia.org/org/jolokia/jolokia_demo").getFullName("another.registry.org"));
         assertEquals("docker.jolokia.org/org/jolokia/jolokia_demo:latest",
                 new ImageName("docker.jolokia.org/org/jolokia/jolokia_demo").getFullName(null));
+        assertEquals("docker.jolokia.org/org/jolokia/jolokia_demo@sha256:2781907cc3ae9bb732076f14392128d4b84ff3ebb66379d268e563b10fbfb9da",
+                new ImageName("docker.jolokia.org/org/jolokia/jolokia_demo@sha256:2781907cc3ae9bb732076f14392128d4b84ff3ebb66379d268e563b10fbfb9da").getFullName(null));
```

Comment:
Could add a check here that ImageName.getRegistry() returns indeed docker.jolokia.org ? (i doubt that)

Reply:
Yep, you're right